### PR TITLE
updating contributing.md and readme.md

### LINKED ISSUE #38: Create a contributing.md file
Would be useful for the future if we have new contributors to have an exact guide how to contribute to this project.

### PR DESCRIPTION
**Pull request by Gongsta**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+## Set up instructions
+
+You're going to have to do those things to contribute to this project.
+
+1. Fork the repo
+2. Clone your fork
+3. Create upstream by running `git remote add upstream https://github.com/KuriusMTL/HackItForward.git`. You will then be able to run `git pull upstream master`to pull any changes from the the original repository since you last forked it).
+4. Create a branch by running `git checkout -b <branch-name>`
+5. Run these commands
```
Comment (fluix-dev) [inline CONTRIBUTING.md]: We should probably mention Python virtual environments.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+```
+pip install -r requirements.txt
+python manage.py migrate
+python manage.py loaddata users
+python manage.py loaddata tags
+python manage.py loaddata social_links
+python manage.py loaddata challenges
+python manage.py loaddata projects
+```
+6. To start the Django server, run `python manage.py runserver`
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
6. To start the Django server, run `python manage.py runserver` and access it at [http://127.0.0.1:8080](http://127.0.0.1:8080)
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+7. Contribute!
+8. If you get things working, add your changed files with `git add .` and run `git commit -m "insert message here"` to commit your messages
+9. Push your changes to your fork with `git push origin <branch-name>`
```
Comment (fluix-dev) [inline CONTRIBUTING.md]: I think this can just be summarized with: "Commit and push your changes."

**Review by fluix-dev — CHANGES_REQUESTED** [review]
In general, I think a CONTRIBUTING file specifies what's specific to this project as opposed to others. While I'm not opposed to explain forking, etc. we should mention what knowledge is needed and include the precise styling we want:
 - Format python files with `black`, cut off lines at 100 characters.
 - Indent Python files with 4 spaces
 - Indent HTML files with 2 spaces.

File: CONTRIBUTING.md (hunk 2 above)
Comment (dulldesk) [reply CONTRIBUTING.md]: Do we need to mention `local_settings.py` (and/for its use in social authentication)?

Comment (fluix-dev) [reply CONTRIBUTING.md]: This shouldn't go into contributing at all tbh, it should be in separate documentation (like an install section of the README).

Comment (fluix-dev) [reply CONTRIBUTING.md]: Also this suggestion is pretty bad, the CLI will tell you where to access it, so don't commit this.

File: CONTRIBUTING.md (hunk 1 above)
Comment (Gongsta) [reply CONTRIBUTING.md]: Can you directly add your suggestions? I will accept the changes you suggest

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+```
+pip install -r requirements.txt
+python manage.py migrate
+python manage.py loaddata users
+python manage.py loaddata tags
+python manage.py loaddata social_links
+python manage.py loaddata challenges
+python manage.py loaddata projects
+```
+6. To start the Django server, run `python manage.py runserver` and access it at [http://127.0.0.1:8080](http://127.0.0.1:8080)
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
6. To start the Django server, run `python manage.py runserver`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+
+### Python Style Guide
+
+All Python code is formatted with [Black](https://github.com/psf/black).
+
+* Cut off lines at 100 characters
+* Indent with 2 spaces
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
* Indent with 2 spaces
* You can run `black .` in the root of the project to format everything for you.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+
+### HTML Style Guide
+
+* Indent with 2 spaces
```
Comment (fluix-dev) [inline CONTRIBUTING.md:36]:
```suggestion
* Indent with 2 spaces
* Indent Jinja template blocks (like `{% if ... %}` as well).
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+
+## Code of Conduct
```
Comment (fluix-dev) [inline CONTRIBUTING.md]: I'm not a huge fan of the Contributor Covenant, especially the multiple specific clauses followed by a more generic catch-all. I'll see if I can write something we all agree on that's shorter and better.

File: README.md [hunk 8]
```diff
@@ -9,17 +9,38 @@
...
+```
+pip install -r requirements.txt
+python manage.py migrate
+python manage.py loaddata users
+python manage.py loaddata tags
+python manage.py loaddata social_links
+python manage.py loaddata challenges
+python manage.py loaddata projects
+```
+2. To start the Django server, run `python manage.py runserver`
```
Comment (fluix-dev) [inline README.md:34]:
```suggestion
2. To start the Django server, run `python manage.py runserver`

**Note**: You should do this in a Python [virtualenv](https://virtualenv.pypa.io/en/stable/) or similar alternative.
```

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Otherwise LGTM. I'll also commit some examples for the style guides in a bit.

File: CONTRIBUTING.md (hunk 7 above)
Comment (Gongsta) [reply CONTRIBUTING.md]: Ok, but for now it will suffice

**Comment by Gongsta** [comment]
I approved the changes, we can merge :))

File: CONTRIBUTING.md (hunk 7 above)
Comment (fluix-dev) [reply CONTRIBUTING.md]: Is there some rush to get this merged? Even if there is, the default should be no code of conduct, not something we'll change within the week.

**Comment by Gongsta** [comment]
@TheAvidDev i added the code of conduct file, updating it with the ruby one

File: CODE_OF_CONDUCT.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+## The HackItForward Community Conduct Guideline
+
+This document provides community guidelines for a safe, respectful, productive, and collaborative place for any person who is willing to contribute to the HackItForward community. It applies to all “collaborative space”, which is defined as community communications channels (such as mailing lists, submitted patches, commit comments, etc.).
```
Comment (fluix-dev) [inline CODE_OF_CONDUCT.md]:
```suggestion
This document provides community guidelines for a safe, respectful, productive, and collaborative place for any person who is willing to contribute to the HackItForward community. It applies to all “collaborative space”, which is defined as community communications channels (such as submitted patches, commit comments, etc.).
```

File: CODE_OF_CONDUCT.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+* Participants will be tolerant of opposing views.
+* Participants must ensure that their language and actions are free of personal attacks and * disparaging personal remarks.
+* When interpreting the words and actions of others, participants should always assume good intentions.
+* Behaviour which can be reasonably considered harassment will not be tolerated.
+
+### Attribution
+
+This Code of Conduct is adapted from the [The Ruby Community Conduct Guideline][homepage].
+
+[homepage]: https://www.ruby-lang.org/en/conduct/
```
Comment (fluix-dev) [inline CODE_OF_CONDUCT.md]:
```suggestion
This Code of Conduct is adapted from the [The Ruby Community Conduct Guideline](https://www.ruby-lang.org/en/conduct/).
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+
+### Python Style Guide
+
+All Python code is formatted with [Black](https://github.com/psf/black).
+
+* Cut off lines at 100 characters
+* Indent with 2 spaces
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
* Indent with 4 spaces
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+When contributing to this repository, please first discuss the change you wish to make via [issues](https://github.com/KuriusMTL/HackItForward/issues).
+
+Please note we have a [Code of Conduct](./CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+## Pull Request Process
+
+1. Fork the repository
+2. Clone your fork
+3. Create a branch by running `git checkout -b <branch-name>`
+4. Contribute!
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
4. Write code!
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,38 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via [issues](https://github.com/KuriusMTL/HackItForward/issues).
+
+Please note we have a [Code of Conduct](./CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
Please follow our [Code of Conduct](./CODE_OF_CONDUCT.md) in all your interactions with the project.
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+
+## Pull Request Process
+
+1. Fork the repository
+2. Clone your fork
+3. Create a branch by running `git checkout -b <branch-name>`
+4. Contribute!
+5. If you get things working, add your changed files with `git add` and run `git commit -m "insert message here"` to commit your messages
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
5. If you get things working, add your changed files with `git add` and then commit them with `git commit`
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -0,0 +1,38 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via [issues](https://github.com/KuriusMTL/HackItForward/issues).
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
When contributing to this repository, first discuss the change you wish to make via [issues](https://github.com/KuriusMTL/HackItForward/issues), if the change is large enough to warrant such discussion.
```

File: CODE_OF_CONDUCT.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
+## The HackItForward Community Conduct Guideline
+
+This document provides community guidelines for a safe, respectful, productive, and collaborative place for any person who is willing to contribute to the HackItForward community. It applies to all “collaborative space”, which is defined as community communications channels (such as mailing lists, submitted patches, commit comments, etc.).
+
+* Participants will be tolerant of opposing views.
+* Participants must ensure that their language and actions are free of personal attacks and * disparaging personal remarks.
```
Comment (fluix-dev) [inline CODE_OF_CONDUCT.md]:
```suggestion
* Participants must ensure that their language and actions are free of personal attacks and disparaging personal remarks.
```

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Otherwise LGTM.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+### Python Style Guide
+
+All Python code is formatted with [Black](https://github.com/psf/black).
+
+* Cut off lines at 100 characters
+* Indent with 4 spaces
+* You can run `black .` in the root of the project to format everything for you.
+
+
```
Comment (fluix-dev) [inline CONTRIBUTING.md]:
```suggestion
```

File: README.md [hunk 18]
```diff
@@ -9,17 +9,40 @@
...
+With the world shifting online as a result of the pandemic, developers are becoming an increasing necessity in any organization. However, during this period, many organizations with limited resources, most often nonprofits, are struggling with this difficult transition. HackItForward was created to help these organizations tackle their problems, by offering up-and-coming developers a platform to work on real-world projects and make a meaningful impact on society.
 
-*This project was made for KuriusHacks 2020 by [TheAvidDev](https://github.com/TheAvidDev/), [dulldesk](https://github.com/dulldesk/), [skyflaren](https://github.com/skyflaren/), and [aavaSapkota](https://github.com/aavaSapkota) where we won the first place award.*
+The driving functionality of HackItForward are the challenges. These come is varying levels of difficulties and requirements. As a general guideline, there are two main types of challenges, tutorial challenges and real-world challenges:
```
Comment (fluix-dev) [inline README.md]:
```suggestion
The driving functionality of HackItForward are the challenges. These come in varying levels of difficulty and requirements. As a general guideline, there are two main types of challenges, tutorial challenges and real-world challenges:
```

**Pull request merged**